toolchains/cc/cc.nix

```diff
@@ -53,239 +54,246 @@ let
-        -fdata-sections
-      )
-      OPT_LINK_FLAGS=(
-        ${
-          if pkgs.stdenv.isDarwin
-          then ""
-          else "$(add_linker_option_if_supported -Wl,--gc-sections -gc-sections)"
-        }
-      )
-      UNFILTERED_COMPILE_FLAGS=(
-        $(
-          if is_compiler_option_supported -fno-canonical-system-headers; then
-            echo -fno-canonical-system-headers
-          elif is_compiler_option_supported -no-canonical-prefixes; then
-            echo -no-canonical-prefixes
-          fi
-        )
-        # Make C++ compilation deterministic. Use linkstamping instead of these
-        # compiler symbols.
-        -Wno-builtin-macro-redefined
-        -D__DATE__=\\\"redacted\\\"
-        -D__TIMESTAMP__=\\\"redacted\\\"
-        -D__TIME__=\\\"redacted\\\"
-      )
-      DBG_COMPILE_FLAGS=(-g)
-      COVERAGE_COMPILE_FLAGS=(
-        ${
-          if pkgs.stdenv.isDarwin then
-            "-fprofile-instr-generate -fcoverage-mapping"
-          else
-            "--coverage"
-        }
-      )
-      COVERAGE_LINK_FLAGS=(
-        ${
-          if pkgs.stdenv.isDarwin then
-            "-fprofile-instr-generate"
-          else
-            "--coverage"
-        }
+      # Removal of unused code and data at link time (can this increase binary
+      # size in some cases?).
+      -ffunction-sections
+      -fdata-sections
+    )
+    OPT_LINK_FLAGS=(
+      ${
+        if pkgs.stdenv.isDarwin
+        then ""
+        else "$(add_linker_option_if_supported -Wl,--gc-sections -gc-sections)"
+      }
+    )
+    UNFILTERED_COMPILE_FLAGS=(
+      $(
+        if is_compiler_option_supported -fno-canonical-system-headers; then
+          echo -fno-canonical-system-headers
+        elif is_compiler_option_supported -no-canonical-prefixes; then
+          echo -no-canonical-prefixes
+        fi
       )
-      SUPPORTS_START_END_LIB=(
-        $(
-          if [[ -x ${cc}/bin/ld.gold ]]; then echo True; else echo False; fi
-        )
+      # Make C++ compilation deterministic. Use linkstamping instead of these
+      # compiler symbols.
+      -Wno-builtin-macro-redefined
+      -D__DATE__=\\\"redacted\\\"
+      -D__TIMESTAMP__=\\\"redacted\\\"
+      -D__TIME__=\\\"redacted\\\"
+    )
+    DBG_COMPILE_FLAGS=(-g)
+    COVERAGE_COMPILE_FLAGS=(
+      ${
+        if pkgs.stdenv.isDarwin then
+          "-fprofile-instr-generate -fcoverage-mapping"
+        else
+          "--coverage"
+      }
+    )
+    COVERAGE_LINK_FLAGS=(
+      ${
+        if pkgs.stdenv.isDarwin then
+          "-fprofile-instr-generate"
+        else
+          "--coverage"
+      }
+    )
+    SUPPORTS_START_END_LIB=(
+      $(
+        if [[ -x ${cc}/bin/ld.gold ]]; then echo True; else echo False; fi
       )
-      IS_CLANG=(
-        $(
-          ${cc}/bin/cc -v 2>&1 | grep -q clang && echo True || echo False
-        )
+    )
+    IS_CLANG=(
+      $(
+        ${cc}/bin/cc -v 2>&1 | grep -q clang && echo True || echo False
```

Comment:
What happened to the static check for `clang` we tried last week?

Reply:
I've pushed it separately in #216 because it essentially changes API.